resolver/auth_mutations.resolvers_test.go

```diff
@@ -555,57 +419,47 @@ func GetSuccessCase() struct {
 			Ok: true,
 		},
 		wantErr: false,
+		init: func(mock sqlmock.Sqlmock) *gomonkey.Patches {
+			rows := sqlmock.NewRows([]string{"id", "email", "password"}).
+				AddRow(testutls.MockID, testutls.MockEmail, OldPasswordHash)
+			mock.ExpectQuery(regexp.QuoteMeta(`select * from "users" where "id"=$1`)).
+				WithArgs().
+				WillReturnRows(rows)
+			result := driver.Result(driver.RowsAffected(1))
+			mock.ExpectExec(regexp.QuoteMeta(`UPDATE "users" `)).WillReturnResult(result)
+			return nil
+		},
+	}
+}
+func loadChangePasswordTestCases() []changePasswordType {
+	return []changePasswordType{
+		changePasswordErrorFindingUserCase(),
+		changePasswordErrorPasswordValidationcase(),
+		changePasswordErrorInsecurePasswordCase(),
+		changePasswordErrorUpdateUserCase(),
+		changePasswordErrorFromConfigCase(),
+		changePasswordSuccessCase(),
 	}
 }
 func TestChangePassword(
 	t *testing.T,
 ) {
 	// Define a struct to represent the change password request
-	cases := GetChangePasswordTestCases()
+	cases := loadChangePasswordTestCases()
 	// Create a new instance of the resolver
 	resolver1 := resolver.Resolver{}
 	for _, tt := range cases {
 		t.Run(
 			tt.name,
 			func(t *testing.T) {
 				// Handle the case where there is an error while loading the configuration
-				if tt.name == ErrorFromConfig {
-					patch := gomonkey.ApplyFunc(config.Load, func() (*config.Configuration, error) {
-						return nil, fmt.Errorf(ErrorMsgFromConfig)
-					})
-					defer patch.Reset()
-				}
-				// Load environment variables
-				err := config.LoadEnv()
-				if err != nil {
-					fmt.Print("error loading .env file")
-				}
-				// Create a mock SQL database connection
 				mock, cleanup, _ := testutls.SetupMockDB(t)
-				// Handle the case where there is an error while finding the user
-				if tt.name == ErrorFindingUser {
-					mock.ExpectQuery(regexp.QuoteMeta(`select * from "users" where "id"=$1`)).
-						WithArgs().
-						WillReturnError(fmt.Errorf(""))
-				}
-				// Expect a query to get the user by ID to return a row with mock data entered
-				rows := sqlmock.NewRows([]string{"id", "email", "password"}).
-					AddRow(testutls.MockID, testutls.MockEmail, OldPasswordHash)
-				mock.ExpectQuery(regexp.QuoteMeta(`select * from "users" where "id"=$1`)).
-					WithArgs().
-					WillReturnRows(rows)
-				// Handle the case where the password update is successful
-				if tt.name == SuccessCase {
-					result := driver.Result(driver.RowsAffected(1))
-					mock.ExpectExec(regexp.QuoteMeta(`UPDATE "users" `)).WillReturnResult(result)
-				}
-				// Handle the case where there is an error while updating the user's password
-				if tt.name == ErrorUpdateUser {
-					mock.ExpectExec(regexp.QuoteMeta(`UPDATE "users" `)).WillReturnError(fmt.Errorf("errrorr"))
-				}
+				defer cleanup()
```

Comment:
if we are deferring the cleanup then it shouldn't be in the loop. We should be able to either pull the `testutls.SetupMockDB(t)` call up, or cleanup at the end of the loop